### ISSUE TITLE
Degenerated gain matrix when using om_gain

### ISSUE THREAD

**Issue by massich** [openmeeg/openmeeg#341]
here's the execution with [this data](https://www.dropbox.com/sh/av40f6brhv5qdxn/AAApYZbe8eC-3-YCV74AvUXNa?dl=0)

```sh
./om_gain -EEG \
          ~/Dropbox/openmeeg_segfault/openmeeg_hminv.mat \
          ~/Dropbox/openmeeg_segfault/openmeeg_dsm.mat \
          ~/Dropbox/openmeeg_segfault/openmeeg_h2em.mat \
          /tmp/gain_eeg.mat

om_gain version 2.4.0 compiled at Aug 17 2018 15:21:43 using OpenMP
 Executing using 4 threads.


| ------ om_gain
| -EEG
| ./openmeeg_hminv.mat
| ./openmeeg_dsm.mat
| ./openmeeg_h2em.mat
| /tmp/openmeeg_gain_eeg.mat
| -----------------------
Using variable with name: matrix
-------------------------------------------
| Elapsed Time: 21.4012 s.
-------------------------------------------

~/Dropbox/openmeeg_segfault 21s
❯ om_matrix_info -i /tmp/gain_eeg.mat                                                       
om_matrix_info version 2.4.0 compiled at Aug 17 2018 15:22:01 using OpenMP
 Executing using 4 threads.

Loading : /tmp/gain_eeg.mat
Dimensions : 257 x 45000
Min Value : -330.452 (124,1212)
Max Value : 170.216 (124,1192)
First Values
13.0735 -9.86959 0.198685 13.0337 -9.88526 
16.1424 -9.24023 10.517 16.2927 -9.54824 
15.7827 -7.38458 15.9536 15.9731 -7.73443 
14.5447 -5.27728 18.6208 14.7053 -5.55507 
12.9091 -3.77995 20.3059 13.0044 -3.98566 
```
The operation seems to go fine but loading the data with matlab shows a degenerated gain matrix:

```matlab
>> load /tmp/openmeeg_gain_eeg.mat
>> figure; imagesc(linop);colorbar
```
![image](https://user-images.githubusercontent.com/7044835/44289573-2d51a980-a275-11e8-944e-8de4efa92886.png)

**Comment by massich** [comment]
the results are replicated here with a different data set https://neuroimage.usc.edu/forums/t/error-while-computing-sources-input-to-svd-must-not-contain-nan-of-inf/7806/11

**Comment by agramfort** [comment]
I can replicate with data in :
https://www.dropbox.com/sh/7kau5it2w1qw69q/AADGC4xiyJ8e2830spOUiyWKa?dl=0

I am looking into it with @papadop

**Comment by papadop** [comment]
OK. First problem is shown here:
https://fr.mathworks.com/help/matlab/import_export/mat-file-versions.html
Matlab v5 files (with option  -v6) do not support objects of size > 2Gb (13446 x 44565 x8 = 4.46 Gb !!).
But the matrix with the bug last thrusday also was >4Gb so I'm a little bit lost. Is it the matlab file format that is limited to 2Gb or the matlab implementation of that format ? In the former case, matio should give an error (and the only solution is to make matlab7.3 file format to work). In the latter, it is a matio internal bug. I'm working on reproducing it.

**Comment by papadop** [comment]
Another question. It has been stated that matio does not build on windows64 with matlab7.3 activeated. What is the problem. Because, this might be the regression between the previous version (which was using mat7.3 files I think) and the current one. In which case, the safer solution is to adopt matlab7.3 file format, so correcting this build problem would be the priority.

**Comment by massich** [comment]
So we are back to https://github.com/tbeu/matio/pull/89

**Comment by massich** [comment]
and also setting up -v7.3 through matio as default would break in matlab2008 which is the minimum version of brainstorm.

**Comment by agramfort** [comment]
@ftadel we need to be able to save large matrices which is not
possible with matlab v5 format.

is it ok for you to only support matlab 7.3 format in the binaries we ship?

**Comment by ftadel** [comment]
File format 7.3 is available since Matlab 2008a, so it would be ok for the backward compatibility we need.

But why don't you simply use your original .bin format?

**Comment by agramfort** [comment]
>
> File format 7.3 is available since Matlab 2008a, so it would be ok for the
> backward compatibility we need.
>
ok great

> But why don't you simply use your original .bin format?
>
well let's not reopen this debate now. File compression does save you disk
space
especially for the DSM matrix and you can read these files on any machine.
Our
bin format is not really documented / standard although I agree it works
when encapsulated
in brainstorm.

**Comment by ftadel** [comment]
> File compression does save you disk space

We don't care much about space here: we don't keep these files in Brainstorm, they are deleted at every Brainstorm startup or new execution of openmeeg.  
If you prefer 7.3 .mat, no problem for me.

**Comment by massich** [comment]
Closing. The issue is solved via #342. We are bumping to 2.4.1 in #344 to facilitate brainstorm update.
